internal/prompt/prompt.go

```diff
@@ -139,6 +119,29 @@ func (p *Prompt) InputAndValidate(title, message string, defaultResponse *string
 		p.out.Notice(output.Emphasize(title))
 	}
 
+	var nonInteractiveResponse *string
+
+	if p.isForced {
+		nonInteractiveResponse = forcedResponse
+		if nonInteractiveResponse == nil {
+			nonInteractiveResponse = defaultResponse
+		}
+		if nonInteractiveResponse == nil {
+			return "", ErrNoForceOption
+		}
+	}
+
+	if !p.isInteractive {
+		nonInteractiveResponse = defaultResponse
+		if nonInteractiveResponse == nil {
+			return "", interactiveInputError(message)
+		}
+	}
+
+	if p.out.Type().IsStructured() {
+		return *nonInteractiveResponse, nil
+	}
```

Comment:
I think this is addressing what we talked about during stand?

Suggest leaving it out of this PR, so we address it all at once in the story I filed. Otherwise this could create some friction in testing as that's not a change Mark is expecting as part of this story.

This goes for all instances of this logic.